### PR TITLE
Pass token boolean to the listitem function.

### PR DESCRIPTION
**Pull request by x13machine**
**Marked version:**

**Markdown flavor:** Markdown.pl|CommonMark|GitHub Flavored Markdown|n/a

## Description

- Passes token boolean to the listitem function. In response to this issue: https://github.com/markedjs/marked/issues/1430
- Also changes "avaialable" to "available" in CONTRIBUTING.md

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

File: lib/marked.js [hunk 1]
```diff
@@ -1258,18 +1258,19 @@ Parser.prototype.tok = function() {
...
       }
 
       while (this.next().type !== 'list_item_end') {
         body += !loose && this.token.type === 'text'
           ? this.parseText()
           : this.tok();
       }
-
-      return this.renderer.listitem(body);
+      return this.renderer.listitem(body, task);
```
Comment (UziTech) [inline lib/marked.js]: Should we also pass checked?

```suggestion
      return this.renderer.listitem(body, task, checked);
```

**Comment by x13machine** [comment]
Do you think passing loose would be a good idea too?

**Comment by UziTech** [comment]
> Do you think passing loose would be a good idea too?

That is really only useful for telling marked how to parse text in the list item.

**Comment by styfle** [comment]
@x13machine Thanks for the PR!

Can you update the docs in `USING_PRO.md` so that listitem api matches the code?

Making a note for myself: this will be a breaking change.

**Comment by UziTech** [comment]
I don't think this will be a breaking change since we are just adding parameters not changing any current parameters

**Comment by UziTech** [comment]
I would say this is a minor version change, which in a beta version is the same as a patch

**Comment by styfle** [comment]
Oh right, we aren't changing the `Renderer.prototype.listitem` to depend on the new args. So I guess that's not a breaking change 🤔

**Pull request merged**